api/ccapi/include/tensor_dim.h

```diff
@@ -21,15 +22,52 @@
 
 #include <bitset>
 #include <vector>
+#include <cstdint>
 
 #ifdef ENABLE_FP16
 #ifdef USE__FP16
 #define _FP16 __fp16
 #else
 #define _FP16 _Float16
 #endif
+#else /* !ENABLE_FP16 */
+#define _FP16 uint16_t /* Keep the FP16 programming interface, but don't allow using it in run-time */
```

Comment:
after this fix, I think we'll need to add test cases to certify every part is well covered.
In the case of tensors, such as follows.

```
Tensor fp32_tensor(..., Tdatatype::FP32);
EXPECT_THROW({ fp32_tensor.getValue<FP16>(1); }, std::invalid_argument);
EXPECT_THROW(
  {
    fp32_tensor.apply(
      (std::function<_FP16(_FP16)>)[&](_FP16 in) { return idx++ % 2; });
  },
  std::invalid_argument);
```

previously, there was no need to check because `_FP16` cannot be used when fp16 is not enabled.
if everyone agrees, I'll work on adding test cases at the tensor level.